### DISCUSSION TITLE
Migration of "sqlalchemy.orm.scoping import instrument, makeprop" to SQLAlchemy 1.4

### DISCUSSION THREAD

**Discussion by valhuber** [sqlalchemy/sqlalchemy#8968] [Usage Questions]
I am trying to migrate [sqlalchemy_auth](https://github.com/dhiltonp/sqlalchemy_auth) to SQLAlchemy version 1.4.

`sqlalchemy_auth` has code importing `instrument, makeprop` in [auth_session](https://github.com/dhiltonp/sqlalchemy_auth/blob/master/sqlalchemy_auth/auth_session.py) such as:

```
from sqlalchemy.orm import Session
from sqlalchemy.orm.scoping import instrument, makeprop
```

I looked but did not see these in the [migration guide](https://docs.sqlalchemy.org/en/14/changelog/migration_20.html).

I did find these in the old code (see screen shot), but cannot find `def makeprop` in the current source.

Looking for pointers on migrating these...

<img width="1411" alt="sa_1_3_makeprop" src="https://user-images.githubusercontent.com/5579631/206933790-aa6c6b68-c395-43e3-a6c9-b5f67ff42e93.png">

**Comment by zzzeek** [comment]
the approach to proxying the session changed in 1.4 however you can use makeprop() and instrument(), just vendor them:

```py
def instrument(name):
    def do(self, *args, **kwargs):
        return getattr(self.registry(), name)(*args, **kwargs)

    return do



def makeprop(name):
    def set_(self, attr):
        setattr(self.registry(), name, attr)

    def get(self):
        return getattr(self.registry(), name)

    return property(get, set_)

```

that's all they are...

**Reply by valhuber** [reply]
I think you the **most** responsive provider I have _ever_ seen, any package, any product.  We sincerely appreciate it.

**Reply by zzzeek** [reply]
it's because I hate my inbox

**Reply by zzzeek** [reply]
less flippantly, what I mean is, I keep things I need to answer in my inbox.  if the inbox grows beyond a day or so of items it just becomes a bigger backlog.   So I try to flatten it to near zero as much as possible.
